src/shogun/machine/KernelMachine.cpp

```diff
@@ -516,12 +509,12 @@ SGVector<float64_t> CKernelMachine::apply_locked_get_output(
 		int32_t start = thread_num * step;
 		int32_t end =
 		    (thread_num == num_threads) ? num_inds : (thread_num + 1) * step;
-#ifdef WIN32
+
 		for (int32_t vec = start; vec < end; vec++)
-#else
-		for (int32_t vec = start; vec < end && !cancel_computation(); vec++)
-#endif
 		{
+#ifdef WIN32
```

Comment:
i'm not so sure if i get why do we have this #ifdef WIN32.... on non-windows we dont wanna allow controllers?

Reply:
@shubham808 Same here.